pkg/booleanpolicy/augmentedobjs/construct.go

```diff
@@ -216,42 +211,41 @@ func ConstructImage(image *storage.Image) (*pathutil.AugmentedObj, error) {
 	}
 
 	if features.ImageSignatureVerification.Enabled() {
-		var addedAtLeastOnSignatureVerificationResult bool
+
 		// Since policies query for image verification status as a single boolean field, we add it to the image here.
 		for i, result := range image.GetSignatureVerificationData().GetResults() {
-			if result.GetStatus() == storage.ImageSignatureVerificationResult_VERIFIED {
-				if result.GetVerifierId() == "" {
-					return nil, utils.Should(errox.InvariantViolation.New(
-						"missing verifier ID in signature verification results"))
-				}
-				err := obj.AddPlainObjAt(
-					&imageSignatureVerification{
-						VerifiedBy: result.GetVerifierId(),
-					},
-					pathutil.FieldStep("SignatureVerificationData"),
-					pathutil.FieldStep("Results"),
-					pathutil.IndexStep(i),
-					pathutil.FieldStep(imageSignatureVerifiedKey),
-				)
-				if err != nil {
-					return nil, utils.Should(err)
-				}
-
-				addedAtLeastOnSignatureVerificationResult = true
+			err := obj.AddPlainObjAt(
+				result.GetVerifierId(),
+				pathutil.FieldStep("SignatureVerificationData"),
+				pathutil.FieldStep("Results"),
+				pathutil.IndexStep(i),
+				pathutil.FieldStep("VerifierId"),
+			)
+			if err != nil {
+				return nil, utils.Should(err)
+			}
+			err = obj.AddPlainObjAt(
+				result.GetStatus().String(),
+				pathutil.FieldStep("SignatureVerificationData"),
+				pathutil.FieldStep("Results"),
+				pathutil.IndexStep(i),
+				pathutil.FieldStep("Status"),
+			)
+			if err != nil {
+				return nil, utils.Should(err)
 			}
 		}
```

Comment:
You do not need to do this. The image signature results field is already a part of image object. Here, you are reading it out and writing it back.

Augmented objects constructions must be used to attach new fields to an objects.